src/project.coffee

```diff
@@ -203,10 +204,12 @@ class Project extends Model
     for provider in @directoryProviders
       break if directory = provider.directoryForURISync?(projectPath)
     if directory is null
-      # This should never happen because DefaultDirectoryProvider should always
-      # return a Directory.
-      throw new Error(projectPath + ' could not be resolved to a directory')
-    @rootDirectories.push(directory)
+      # DefaultDirectoryProvider doesn't handle specific directory protocols
+      # Atom Packages may use packageDirectoryPaths to add thier own paths later.
```

Comment:
So you are saying we can misuse the directoryProvider to handle protocols
like `http://`?

On Fri, Feb 27, 2015 at 5:03 PM, Wliu <notifications@github.com> wrote:

> In src/project.coffee
> <https://github.com/atom/atom/pull/5795#discussion_r25551228>:
>
> > @@ -203,10 +204,12 @@ class Project extends Model
> >      for provider in @directoryProviders
> >        break if directory = provider.directoryForURISync?(projectPath)
> >      if directory is null
> > -      # This should never happen because DefaultDirectoryProvider should always
> > -      # return a Directory.
> > -      throw new Error(projectPath + ' could not be resolved to a directory')
> > -    @rootDirectories.push(directory)
> > +      # DefaultDirectoryProvider doesn't handle specific directory protocols
> > +      # Atom Packages may use packageDirectoryPaths to add thier own paths later.
>
> their
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/pull/5795/files#r25551228>.
>